### ISSUE TITLE
Is it possible to add CSS hooks on <amp-list> ?

### ISSUE THREAD

**Issue by meistudioli** [ampproject/amphtml#15478]
To whom it may concern:

I like to add some vivid effect on **&lt;amp-list&gt;** and wondering is it possible to add some CSS hooks on it, just like &lt;amp-form&gt; ?

such as:
* .amp-list-fetching
* .amp-list-fetch-success
* .amp-list-fetch-error

Thanks for replying.

Paul

**Comment by meistudioli** [comment]
Hi, @aghassemi :

I like to do some "infinite scroll" feature and try to use the advantage of amp-list (fetch data when in in viewport). So I can add a trigger on next amp-list with CSS hooks.

Here comes some sample code:
```
&lt;style amp-custom&gt;
amp-list.amp-list-fetch-success+amp-list{display:block;}
&lt;/style&gt;

&lt;amp-list class="amp-list-fetch-success"&gt;
	....
&lt;/amp-list&gt;

&lt;amp-list&gt;
	....
&lt;/amp-list&gt;

&lt;amp-list&gt;
	....
&lt;/amp-list&gt;
```

Paul

**Comment by cathyxz** [comment]
Hi @meistudioli does the current `amp-list-load-more` feature in https://github.com/ampproject/amphtml/issues/14060 solves your use case?

**Comment by meistudioli** [comment]
Hi, @cathyxz:

Actually NO, but I've already solve feature in different way. Still wondering it still can't add CSS hook when status change ?

**Comment by cathyxz** [comment]
So implementing this isn't difficult, but like @aghassemi said previously, could introduce content shifting issues if your CSS changes the size of `<amp-list>`. The way I understand your feature request (feel free to correct me if wrong) is that you would like ways to visually distinguish between the following states of amp-list: 
1. Fetch in progress
2. Fetch failed
3. Fetch succeeded

We currently offer the following conventions for visually distinguishing these states: 
1. Fetch in progress
- We have a loading indicator and a placeholder element, which is customizable via a child element `<div placeholder>` within `<amp-list>`. While a fetch is in progress, this element is shown. 
2. Fetch failed
- We have a fallback element specified as the child element `<div fallback>` within `<amp-list>`. If a fetch failed, this element is shown. 
3. Fetch succeeded
- We assume that this is the default case, so you can just apply default styles here. In this case, the successfully fetched items will be shown. 

Could you tell me more about why this (or the `amp-list-load-more` feature) isn't working for your use case? The main challenges around your feature request are making them consistent with our layout restrictions and constraints against content-shifting.

**Comment by meistudioli** [comment]
Hi, @cathyxz 

Here comes my another use case. amp-list require [layout] before we implement it. That means it has already occupied space in web page. Once web service response non 2xx, it will be a big blank in the page. That why I try to hide it when fetch fail occurred.

**Comment by cathyxz** [comment]
That makes a lot of sense. I think somewhere else related to amp-list, I recall seeing a "collapse amp-list on fetch fail" feature request, which seems to be along the same lines. While I don't want to commit to an implementation strategy just yet, we'll keep this in mind for V2. 

Thank you for the feedback!

**Comment by cathyxz** [comment]
Actually, upon thinking through this a bit more, there's one condition here. We can consider collapsing components if they are outside of the viewport, but definitely not if they are already visible in the viewport, since that would cause content shifting and violate our core UX principles. So depending on what exactly you are envisioning, I'm not 100% certain that your use case is something we can support.

**Comment by aghassemi** [comment]
deduping with https://github.com/ampproject/amphtml/issues/12212
